docs/SPEC.md

```diff
@@ -2064,37 +2064,72 @@ from(bucket:"telegraf/autogen")
 ```
 #### Group
 
-Group groups records based on their values for specific columns.
+_Group_ groups records based on their values for specific columns.
 It produces tables with new group keys based on the provided properties.
 
-Group has the following properties:
+Group accepts _exactly one_ of the following properties:
 
 *  `by` list of strings
     Group by these specific columns.
-    Cannot be used with `except`.
 *  `except` list of strings
     Group by all other columns except this list.
-    Cannot be used with `by`.
+*  `none` boolean
+    Group by nothing, so, group all records into a single group.
+*  `all` boolean
+    Group by every column.
+    
+Specifying more than one of the properties above would result in an error.
 
-Examples:
-    group(by:["host"]) // group records by their "host" value
-    group(except:["_time", "region", "_value"]) // group records by all other columns except for _time, region, and _value
-    group(by:[]) // group all records into a single group
-    group(except:[]) // group records into all unique groups
+__Examples__
+
+_By_
 
 ```
 from(bucket: "telegraf/autogen") 
     |> range(start: -30m) 
     |> group(by: ["host", "_measurement"])
 ```
-All records are grouped by the "host" and "_measurement" columns. The resulting group key would be ["host, "_measurement"]
+
+Records are grouped by the `"host"` and `"_measurement"` columns.  
+The resulting group key is `["host", "_measurement"]`, so a new table for every different `["host", "_measurement"]`
+value is created.  
+Every table in the result contains every record for some `["host", "_measurement"]` value.  
+Every record in some resulting table has the same value for the columns `"host"` and `"_measurement"`.
+
+_Except_
 
 ```
 from(bucket: "telegraf/autogen")
     |> range(start: -30m)
     |> group(except: ["_time"])
 ```
-All records are grouped by the set of all columns in the table, excluding "_time". For example, if the table has columns ["_time", "host", "_measurement", "_field", "_value"] then the group key would be ["host", "_measurement", "_field", "_value"]
+
+Records are grouped by the set of all columns in the table, excluding `"_time"`.  
+For example, if the table has columns `["_time", "host", "_measurement", "_field", "_value"]` then the group key would be
+`["host", "_measurement", "_field", "_value"]`.
+
+_None_
+
+```
+from(bucket: "telegraf/autogen")
+    |> range(start: -30m)
+    |> group(none: true)
+```
+
+Records are grouped into a single table.  
+The group key of the resulting table is empty.
+
+_All_
+
+```
+from(bucket: "telegraf/autogen")
+    |> range(start: -30m)
+    |> group(all: true)
+```
+
+Group records by every column.  
```

Comment:
@stuartcarnie 

As you can see, I opted for different levels of abstraction in the description.